libs/vkd3d/command.c

```diff
@@ -6403,6 +6414,35 @@ static void STDMETHODCALLTYPE d3d12_command_list_CopyResource(d3d12_command_list
     VKD3D_BREADCRUMB_COMMAND(COPY);
 }
 
+static void d3d12_command_list_end_batch(struct d3d12_command_list *list)
+{
+    switch (list->transfer_batch.batch_type)
+    {
+        case VKD3D_BATCH_TYPE_COPY_BUFFER_TO_IMAGE:
+        case VKD3D_BATCH_TYPE_COPY_IMAGE_TO_BUFFER:
+        case VKD3D_BATCH_TYPE_COPY_IMAGE:
+            d3d12_command_list_end_current_render_pass(list, false);
```

Comment:
Could just skip this if batch_len is 0. This will be called unconditionally in many places.

Reply:
The batch_type would be `NONE` in such cases. Does separating that test out of `switch` sounds good?

Reply:
That is true ... Just keep it as is then.